### DISCUSSION TITLE
Catalyst 21.12

### DISCUSSION THREAD

**Discussion by Scitator** [catalyst-team/catalyst#1388] [Announcements]
### Tl;dr

Distributed engines update (multi-node support) and many other improvements.

### Added

- MNIST dataset for SSL banchmark ([#1368](https://github.com/catalyst-team/catalyst/pull/1368))
- MoveiLens 20M dataset [#1336](https://github.com/catalyst-team/catalyst/pull/1336)
- logger property for logging customization ([#1372](https://github.com/catalyst-team/catalyst/pull/1372))
- MacridVAE example ([#1363](https://github.com/catalyst-team/catalyst/pull/1363))
- SSL benchmark results ([#1374](https://github.com/catalyst-team/catalyst/pull/1374))
- Neptune example ([#1377](https://github.com/catalyst-team/catalyst/pull/1377))
- multi-node support for engines ([#1364](https://github.com/catalyst-team/catalyst/pull/1364))

### Changed

- RL examples update to last version ([#1370](https://github.com/catalyst-team/catalyst/pull/1370))
- DDPLoaderWrapper updated to new version ([#1385](https://github.com/catalyst-team/catalyst/pull/1385))
- `num_classes` for classification metrics became optional ([#1379](https://github.com/catalyst-team/catalyst/pull/1379))
- colab ci/cd update to new verion

### Removed

-

### Fixed

- `requests` requirements for `catalyst[cv]` added ([#1371](https://github.com/catalyst-team/catalyst/pull/1370))
- loader step counter ([#1374](https://github.com/catalyst-team/catalyst/pull/1374))
- detection example data preprocessing ([#1369](https://github.com/catalyst-team/catalyst/pull/1369))
- gradient clipping with fp16 runs ([#1378](https://github.com/catalyst-team/catalyst/pull/1378))
- config API fix for DDP runs ([#1383](https://github.com/catalyst-team/catalyst/pull/1383))
- checkpoint creation for fp16 engines ([#1382](https://github.com/catalyst-team/catalyst/pull/1382))

### Contributors ❤️

@bagxi @ditwoo @MrNightSky @Nimrais @y-ksenia @sergunya17 @Thiefwerty @zkid18

<hr /><em>This discussion was created from the release <a href='https://github.com/catalyst-team/catalyst/releases/tag/v21.12'>Catalyst 21.12</a>.</em>
